### DISCUSSION TITLE
Shipping labels - print in bulk

### DISCUSSION THREAD

**Discussion by t0is** [allegro/allegro-api#7670] [Suggestions - resolved]
According to the documentation, it is only possible for one parcel at a time. 

This is a bit weird and unsual - it always prints on the first 1/4 of the page (when printing to A4 format). 
This basically means that we are able to print one label, turn the page around, print another one. 

And since the API also does not support offset for the labels, there will always be two labels I am not able to print at  

```
.................. |....................
PRINTED............|........NOT....
...................|....................
...................|....................
---------------------------------------
...................|....................
..NOT..............|...PRINTED
...................|....................
...................|....................
```
                    
NOT represents the area that is NOT ACCESSIBLE at all with this logic.

This is both very inconvenient and slow and not very ECO friendly, since the waste this would create is half the labels purchased, which is ridiculous. 

Some of the companies (Packeta for example) also provides a function to shift the offset of the labels on the A4, therefore you have as low waste as possible. 

**Please add an option to print multiple labels at once, not one at a time - More so when this "feature" is already present when printing the labels manually!**

**Comment by Ondrakrejci91** [comment]
How come it´s not been added yet???

**Comment by KLU0116** [comment]
I can't wait for them to add something like this, it's needed !!

**Comment by hahyes** [comment]
@t0is As much I like this idea, I'd like to inform you labels are not always created by Allegro themselfs. Sometimes it depends on what shipment company provides through API integration or it depends on what settings are binded to your shipment service provider account/token. In case of integrations they are simply caching label downloaded from the delivery service company API and returning it to you. 
I don't know, how it works in Packeta to be honest yet it is not simple matter for Allegro to do.

**Reply by t0is** [reply]
Hi @hahyes 
Thanks for your input. 

I understand that they are somewhat dependent on the label provided by the carrier, therefore I would expect a possible struggle with printing, lets say, 4 labels from 4 different carriers at once. 

But I never encountered a carrier API that would not support this multi-print option, therefore at least 4 labels from a single carrier should be possible to print at once. More so when the carrier is Allegro One (I know it is UPS in the background, but they definitelly have this option in their API, since I implemented their API as well in past.)

Second thing - As I stated, I was always able to make this work via manual print on allegro website, that's why I believe it should not be much of an issue to implement also in the API. 

About Packeta - they get an original label from the carrier and just combine them all together into one PDF / ZPF file for you that you can get via API, which is very convenient.

**Reply by hahyes** [reply]
@t0is I just have bad experiences with polish post carrier services (Poczta Polska), where you needed to correct settings in their own panel (not in Allegro). My point is, standardization of this solution might be not easy. Some carriers let you change settings or adjust print, some do not. And as much I know Allegro they won't go into carrier specific solutions in API. It is not perfect and sometimes it leads to really ugly fixes (like support for some carrier services by adding additional service to the package, that was horrible and not documented at all).

**Reply by t0is** [reply]
@hahyes Understandable for sure, but in that case, they should not require us sellers to have Allegro One as a delivery option (Otherwise you won't be able to offer Allegro SMART from 25th October)

All the other required options can be used with your own contract - therefore use you own API implementation, where you can deal with the multi-label print yourself. And I have to say, all the other carriers have a very convenient API in this regard. 

With Allegro One, we are stuck with what allegro provides and right now, it's just not suitable for extensive use. 

The bare minimum in my opinion would be to make the API usable at least for their "own" carrier, Allegro One.

If Allegro is not willing to do that, sadly I feel like it is screwing us sellers by having such requirements.

**Comment by KutH83** [comment]
it's needed !!

**Comment by MartaNowaczyk** [comment]
@t0is When you use the new "[Ship with Allegro](https://developer.allegro.pl/tutorials/how-to-manage-parcels-via-ship-with-allegro-ZM9YAyGKWTV)" resources - [POST /shipment-management/protocol](https://developer.allegro.pl/documentation/#operation/getShipmentProtocol), it is possible to print multiple labels at once. A6 format labels are also supported. Alternatively, PDF readers allow you to print multiple A6 labels on one page. The mentioned option should be supported on the side of external software. 
Due to the implementation of the option to print multiple labels, I mark the suggestion as resolved.
